invenio_rdm_records/services/communities/service.py

```diff
@@ -37,10 +54,109 @@ def record_cls(self):
+                    dsl.Q("term", **{"receiver.community": community_id}),
+                    dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+                    dsl.Q("term", **{"type": CommunityInclusion.type_id}),
+                    dsl.Q("term", **{"is_open": True}),
+                ],
+            ),
+        )
+        return next(results.hits)["id"] if results.total > 0 else None
+
+    def _include(self, identity, community_id, record, uow):
+        """Create request to add the community to the record."""
+        already_included = community_id in record.parent.communities
+        if already_included:
+            raise CommunityAlreadyExists()
+
+        # check if the community exists
+        community = current_communities.service.record_cls.pid.resolve(community_id)
+
+        # check if there is already an open request, to avoid duplications
+        existing_request_id = self._exists(identity, str(community.id), record)
+        if existing_request_id:
+            raise OpenRequestAlreadyExists(existing_request_id)
+
+        type_ = current_request_type_registry.lookup(CommunityInclusion.type_id)
+        receiver = ResolverRegistry.resolve_entity_proxy(
+            {"community": community_id}
+        ).resolve()
+
+        empty_data = {}
+        request_item = current_requests_service.create(
+            identity,
+            empty_data,
+            type_,
+            receiver,
+            topic=record,
+            uow=uow,
+        )
+        # create review request
+        request_item = current_rdm_records.community_inclusion_service.submit(
+            identity, record, community, request_item._request, empty_data, uow
+        )
+        # include directly when allowed
+        return current_rdm_records.community_inclusion_service.include(
+            identity, community, request_item._request, uow
+        )
+
     @unit_of_work()
-    def create(self, identity, data, record, uow=None):
+    def add(self, identity, id_, data, uow):
         """Include the record in the given communities."""
-        pass
+        valid_data, errors = self.schema.load(
+            data,
+            context={
+                "identity": identity,
+                "max_number": self.config.max_number_of_additions,
+            },
+            raise_errors=True,
+        )
+        communities = valid_data["communities"]
+
+        record = self.record_cls.pid.resolve(id_)
+        self.require_permission(identity, "add_community", record=record)
+
+        success = []
+        for community in communities:
+            community_id = community["id"]
+
+            result = {
+                "community": community_id,
+            }
+            try:
+                request_item = self._include(identity, community_id, record, uow)
+                result["request"] = str(request_item.data["id"])
+                success.append(result)
+            except (NoResultFound, PIDDoesNotExistError):
```

Comment:
readablity could be maybe improved if you would consider handlers for these exception types